apps/server/src/modules/learnroom/repo/mikro-orm/course.repo.ts

```diff
@@ -44,4 +47,40 @@ export class CourseMikroOrmRepo extends BaseDomainObjectRepo<Course, CourseEntit
 
 		return courses;
 	}
+
+	public async getCourseInfo(filter: CourseFilter, options?: IFindOptions<Course>): Promise<Page<Course>> {
+		const scope: CourseScope = new CourseScope();
+		scope.bySchoolId(filter.schoolId);
+		if (filter.courseStatusQueryType === CourseStatusQueryType.CURRENT) {
```

Comment:
Having this abstraction in the scope itself would be better, like `scope.byStatus(filter.courseStatusQueryType);`